tests/product/__init__.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import os
+
+
+def determine_jdk_directory(cluster):
```

Comment:
It'll get back all directories that match, it will then separate all components of all those directories and pick the last one as the JDK to use.

Hopefully this shouldn't be a problem since we control the configurable cluster and docker environments so we know what JDKs are installed. I would do a check in this method for the number of matches and if there are more than 1 then I can fail fast. I think that's a good solution. What do you think?